tests/sharingTest.js

```diff
@@ -1,224 +1,538 @@
+// TODO: Unskip the skipped tests after the issue is resolved
+// https://github.com/pact-foundation/pact-js/issues/604
 import { MatchersV3, XmlBuilder } from '@pact-foundation/pact/v3'
 
+const { fromProviderState } = MatchersV3
+const { convertToFormUrlEncodedData } = require('./helpers/sharingHelper')
+
 describe('Main: Currently testing file/folder sharing,', function () {
   const config = require('./config/config.json')
 
   const {
     applicationXmlResponseHeaders,
     applicationFormUrlEncoded,
-    accessControlAllowHeaders,
-    accessControlAllowMethods,
-    validAdminAuthHeaders,
-    xmlResponseHeaders,
+    htmlResponseHeaders,
     getCapabilitiesInteraction,
     getCurrentUserInformationInteraction,
     createOwncloud,
     ocsMeta,
     shareResponseOcsData,
     createProvider,
-    origin
+    getAuthHeaders
   } = require('./pactHelper.js')
 
   // TESTING CONFIGS
   const testUserPassword = config.testUserPassword
-  const testUser = config.testUser
   const testGroup = config.testGroup
-  const testFolder = '/' + config.testFolder
+  const testFolder = config.testFolder
   const nonExistentFile = config.nonExistentFile
   const expirationDate = config.expirationDate
   const testFiles = config.testFiles
 
+  const sharer = config.testUser
+  const sharerPassword = config.testUserPassword
+  const sharee = config.testUser2
+  const shareePassword = config.testUser2Password
+
+  const validAuthHeaders = {
+    authorization: getAuthHeaders(sharer, sharerPassword)
+  }
```

Comment:
may be we can shift this in pactHelper and can use in other files as well, as most of them require validUserAuthHeaders now.

Reply:
we can implement something like this
```js
const getValidUserAuthHeaders = (username, password) => ({
  authorization: getAuthHeaders(username, password)
})
```
and use it like
```js
// on every interaction
...
headers: getValidUserAuthHeaders(username, password)
...

OR

validUserAuthHeaders = getValidUserAuthHeaders(username, password)
...
headers: validUserAuthHeaders
...
```

It is also possible to introduce some confusion between `getAuthHeaders` and `getValidUserAuthHeaders`, and workarounds